source/core_http_client.c

```diff
@@ -553,6 +569,28 @@ static uint32_t getZeroTimestampMs( void )
 
 /*-----------------------------------------------------------*/
 
+static int stringCaseCmp( const char * str1,
+                          const char * str2,
+                          size_t n )
+{
+    for( ; n > 0; n--, str1++, str2++ )
+    {
+        if( toupper( *str1 ) < toupper( *str2 ) )
```

Comment:
This will lead to segfault because you could dereference NULL pointer. We should
```
assert( str1 != NULL );
assert( str2 != NULL );
```
Although, one thing to consider is that maybe we'd like to instead return 0 if both of them are `NULL`.